spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ClassNLLCriterion.scala

```diff
@@ -111,7 +120,12 @@ class ClassNLLCriterion[@specialized(Float, Double) T: ClassTag]
           if (curTarget == -1) (ev.zero, ev.one)
           else {
             val curWeight = if (weights != null) weights.valueAt(curTarget) else ev.fromType[Int](1)
-            (ev.times(input.valueAt(_i, curTarget), curWeight), curWeight)
+            if (acceptProb) {
```

Comment:
The name of the parameter `acceptProb` seems a bit confusing?
If true, `loss = -log(x[class])`
Else, `loss = -x[class]`
Users may find ambiguous for this parameter? Since if it is false, we may assume that they are actually pass in probabilities but we don't accept probabilities?
Also, after adding the log, we assume x[class] is always positive? What if it is zero?

Reply:
@hkvision I'll try to handle the zero case and what's your suggestions about the parameter name?

Reply:
@yangw1234 It seems in keras, when get zero, it will use precision probably treat it as ~1e-7 and calculate the result as the following:

>>> y_true = 3
>>> y_pred = np.array([0.1, 0.6, 0.3, 0])
>>> res = loss(K.variable(y_true),K.variable(y_pred))
>>> K.eval(res)
array([ 16.1180954], dtype=float32)

Do we have the same or similar mechanism in BigDL for this?

Reply:
@yangw1234 Also, in BigDL index starts from 1 and in Keras index starts from 0, will this be a problem?

Reply:
@yangw1234 For the name, I'm not sure lol. Maybe something like takeLog...?

Reply:
@hkvision For the zero probability case, I'll clip the probability to make it between 0+epsilon and 1 - epsilon where epsilon is 1e-8. This is exactly how keras tensorflow backend handle it.
I think index inconsistence should be handled by the keras-bigdl converter.